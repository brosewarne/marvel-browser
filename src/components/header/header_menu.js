import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Button from 'material-ui/Button';
import Menu, { MenuItem } from 'material-ui/Menu';
import Typography from 'material-ui/Typography';

class HeaderMenu extends React.Component {
    static propTypes = {
        menuName: PropTypes.string.isRequired,
        menuOptions: PropTypes.array.isRequired
    };

    constructor(props) {
        super(props);

        this.state = {
            anchorEl: null
        };

        this.menuItems = this.getMenuItems();
    }

    /**
     * Open the menu when the menu button is clicked
     * @param {Object} event - The click event
     */
    onClick = (event) => {
        this.setState({ anchorEl: event.currentTarget });
    };

    /**
     * Close the menu when an option is clicked
     */
    onMenuOptionClick = () => {
        this.setState({ anchorEl: null });
    };

    /**
     * Create MenuItem components from this.props.menuOptions
     * @returns {Array} - The array of MenuItem components
     */
    getMenuItems = () => {
        return this.props.menuOptions.map(
            (menuOption) => {
                return (
                    <MenuItem
                        onClick={this.onMenuOptionClick}
                    >
                        <Link
                            href={menuOption.href}
                            to={{ pathname: menuOption.href }}
                        >
                            <Typography variant="subheading" color="primary" align="center">
                                {menuOption.name}
                            </Typography>
                        </Link>
                    </MenuItem>
                );
            }
        );
    };

    render() {
        const { anchorEl } = this.state;

        return (
            <Fragment>
                <Button
                    aria-owns={anchorEl ? 'simple-menu' : null}
                    aria-haspopup="true"
                    onClick={this.onClick}
                    color="secondary"
                    variant="raised"
                >
                    {this.props.menuName}
                </Button>
                <Menu
                    id="simple-menu"
                    anchorEl={anchorEl}
                    open={Boolean(anchorEl)}
                    onClose={this.handleClose}
                >
                    {this.menuItems}
                </Menu>
            </Fragment>
        );
    }
}

export { HeaderMenu };
