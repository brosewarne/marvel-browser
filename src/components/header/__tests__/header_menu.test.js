import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { HeaderMenu } from '../header_menu';

configure({ adapter: new Adapter() });

describe('<HeaderMenu />', () => {
    it('renders correctly', () => {
        const wrapper = shallow(
            <HeaderMenu menuName="ManuName" menuOptions={[{ href: '/ahref', name: 'option1' }, { href: '/bhref', name: 'option2' }]} />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
