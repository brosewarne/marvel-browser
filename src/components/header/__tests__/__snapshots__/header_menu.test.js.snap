// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderMenu /> renders correctly 1`] = `
<React.Fragment>
  <WithStyles(Button)
    aria-haspopup="true"
    aria-owns={null}
    color="secondary"
    onClick={[Function]}
    variant="raised"
  >
    ManuName
  </WithStyles(Button)>
  <WithStyles(Menu)
    anchorEl={null}
    id="simple-menu"
    open={false}
  >
    <WithStyles(MenuItem)
      onClick={[Function]}
    >
      <Link
        href="/ahref"
        replace={false}
        to={
          Object {
            "pathname": "/ahref",
          }
        }
      >
        <WithStyles(Typography)
          align="center"
          color="primary"
          variant="subheading"
        >
          option1
        </WithStyles(Typography)>
      </Link>
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      onClick={[Function]}
    >
      <Link
        href="/bhref"
        replace={false}
        to={
          Object {
            "pathname": "/bhref",
          }
        }
      >
        <WithStyles(Typography)
          align="center"
          color="primary"
          variant="subheading"
        >
          option2
        </WithStyles(Typography)>
      </Link>
    </WithStyles(MenuItem)>
  </WithStyles(Menu)>
</React.Fragment>
`;
